pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
-            acc_cls = IPUAccelerator
-        else:
-            acc_cls = CPUAccelerator
-
-        accelerator = acc_cls()
-        return accelerator
-
-    def select_cluster_environment(self) -> ClusterEnvironment:
-        if self._cluster_environment is not None:
-            return self._cluster_environment
-        if self._is_slurm_managing_tasks():
-            rank_zero_info("Multiprocessing is handled by SLURM.")
-            return SLURMEnvironment()
-
-        for env_type in (TorchElasticEnvironment, KubeflowEnvironment, LSFEnvironment):
-            if env_type.detect():
-                return env_type()
 
-        return LightningEnvironment()
-
-    def set_distributed_mode(self, strategy: Optional[str] = None):
+    def _lazy_init_strategy(self):
+        # set strategy properties
+        self.strategy.accelerator = self.accelerator
+        if self.precision_plugin:
+            self.strategy.precision_plugin = self.precision_plugin
+        if self.checkpoint_io:
+            self.strategy.checkpoint_io = self.checkpoint_io
+        if hasattr(self.strategy, "cluster_environment"):
+            self.strategy.cluster_environment = self.cluster_environment
+        if hasattr(self.strategy, "parallel_devices"):
+            if self.strategy.parallel_devices:
+                self._parallel_devices = self.strategy.parallel_devices
+            else:
+                self.strategy.parallel_devices = self._parallel_devices
+        if hasattr(self.strategy, "num_nodes"):
+            self.strategy._num_nodes = self._num_nodes_flag
+        if hasattr(self.strategy, "sync_batchnorm"):
+            self.strategy.sync_batchnorm = self.sync_batchnorm
+        if hasattr(self.strategy, "set_world_ranks"):
+            self.strategy.set_world_ranks()
 
-        if strategy is None and self.is_training_type_in_plugins:
-            return
+        from pytorch_lightning.utilities import _IS_INTERACTIVE
 
-        if strategy is not None and strategy in StrategyRegistry:
-            self.distributed_backend = StrategyRegistry[strategy]["distributed_backend"]
-        elif strategy is not None:
-            self.distributed_backend = strategy
+        interactive_compatible_strategy = ("dp", "ddp_spawn", "ddp_sharded_spawn", "tpu_spawn")
+        if _IS_INTERACTIVE and self.strategy.distributed_backend not in interactive_compatible_strategy:
+            raise MisconfigurationException(
+                f"`Trainer(strategy={self.strategy.distributed_backend!r})` or"
+                f" `Trainer(accelerator={self.strategy.distributed_backend!r})` is not compatible with an interactive"
+                " environment. Run your code as a script, or choose one of the compatible backends:"
+                f" {', '.join(interactive_compatible_strategy)}."
+                " In case you are spawning processes yourself, make sure to include the Trainer"
+                " creation inside the worker function."
+            )
```

Comment:
this check can also can be moved to the individual strategies in the init

Reply:
Will add a todo, address in separate PR to limit this PR size

Reply:
Added todo, also tracked in follow up items in #11449 item 5